File: algorithm/121. Best Time to Buy and Sell Stock/121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minprice, maxpro=2147483647,0
        for i in range(len(prices)):
            minprice = min(minprice, prices[i])
            maxpro = max(maxpro, prices[i]-minprice)
        return maxpro            

File: algorithm/121. Best Time to Buy and Sell Stock/121.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxp = 0, minprice = Integer.MAX_VALUE;
        for(int i = 0; i < prices.length; i++)
        {
            minprice = Math.min(minprice, prices[i]);
            maxp = Math.max(maxp, prices[i] - minprice);
        }
        return maxp;
    }
}
